### DISCUSSION TITLE
Using dynamic urls using _sitemap-urls.ts with i18n?

### DISCUSSION THREAD

**Discussion by notflip** [harlan-zw/nuxt-seo#167] [Q&A] [answered]
In my previous (single-language) products, I used `_sitemap-urls.ts` to get all the collections from my CMS and map them for the sitemap, now I'm using Nuxt-I18N, I'm wondering what the multi-language use of https://nuxtseo.com/sitemap/guides/dynamic-urls is? 

Thanks!

**Comment by harlan-zw** [comment]
You can either return them already localised or use i18nTransform util.

See https://nuxtseo.com/sitemap/integrations/i18n#dynamic-urls-with-i18n-using-_i18ntransform

**Reply by notflip** [reply]
Thanks @harlan-zw I suppose if the slugs are also different per language, the only option is to define them manually?

**Comment by notflip** [comment, marked as answer]
Answering my own question here, in case anyone needs it. The following is working now, I'm using the returned language code from the CMS to determine which sitemap the content goes in.

```
results.push({
    loc: `/cases/${caseItemTranslation.slug}`,
    lastmod: new Date(caseItem.date_updated).toISOString(),
    _sitemap: caseItemTranslation.languages_code,
})
```

**Reply by harlan-zw** [reply]
Looks good, you may consider adding alternatives though

**Reply by notflip** [reply]
@harlan-zw How do you mean alternatives?
